include/subprocess/subprocess.hpp

```diff
@@ -102,6 +89,39 @@ class command_error : public subprocess_error
 };
 } // namespace exceptions
 
+/**
+ * @brief Raise a std::system_error when a syscall fails
+ *
+ * @param name The name of the offending function.
+ * @param errval The errno-style error code.
+ * @param extra Additional text to display in the error message.
+ */
+[[noreturn]] void raise_sys_error(std::string name, int errval, const std::string& extra = "")
+{
+  if (extra.length())
+  {
+    name += ": ";
+    name += extra;
+  }
+  throw std::system_error(errval, std::generic_category(), name);
+}
+
+/**
+ * @brief Raise a std::system_error when a syscall fails
+ *
+ * @param name The name of the offending function.
+ * @param extra Additional text to display in the error.
+ */
+[[noreturn]] void raise_sys_error(std::string name, const std::string& extra = "")
```

Comment:
If `extra` is removed, and `errval` is defaulted to `errno`, this can effectively be combined into above.